File: typescript/1957.DeleteCharacterstoMakeFancyString.ts
/**
 * Deletes characters from a string to make it a fancy string.
 * @param {string} s - The input string.
 * @returns {string} - The resulting fancy string.
 */
function makeFancyString(s: string): string {
    const result: string[] = [];
    for (const char of s) {
        if (
            result.length >= 2 &&
            result[result.length - 1] === result[result.length - 2] &&
            char === result[result.length - 1]
        ) continue;
        result.push(char);
    }
    return result.join('');
};

File: python/easy/1957-delete-characters-to-make-fancy-string.py
class Solution:
    def makeFancyString(self, s: str) -> str:
        """
        Deletes characters from a string to make it a fancy string.

        Args:
          s (str): The input string.

        Returns:
          str: The resulting fancy string.
        """
        result = []
        for char in s:
            if len(result) >= 2 and result[-1] == result[-2] == char:
                continue
            result.append(char)
        return "".join(result)
